OnlineLibraryDB/StoredProcedures/Books/sp_UpdateBookAndReservations.sql

```diff
@@ -0,0 +1,44 @@
+﻿CREATE PROCEDURE [dbo].[sp_UpdateBookAndReservations]
+	@reservations [dbo].[t_Reservation] READONLY,
+	@name nvarchar(250),
+	@description nvarchar(MAX),
+	@inArchive bit,
+	@genre tinyint,
+	@bookId int
+AS
+BEGIN
+	SET NOCOUNT ON;
+
+	BEGIN TRAN
+
+		UPDATE [dbo].[Books] 
+		SET Name = @name, Description=@description, InArchive=@inArchive, Genre=@genre 
+		WHERE Id = @bookId;
+
+		IF NOT EXISTS (SELECT 1 FROM @reservations)
```

Comment:
Do you need this check?
Does it not covered in following? 
`-- delete all rows where Id not in select
			DELETE FROM [dbo].[Reservations] 
			WHERE Id NOT IN (SELECT Id FROM @reservations) AND BookId = @bookId;`